packages/bsky/src/hydration/hydrator.ts

```diff
@@ -517,18 +528,21 @@ export class Hydrator {
     })
     // hydrate state for all posts, reposts, authors of reposts + reply parent authors
     const repostUris = mapDefined(items, (item) => item.repost?.uri)
-    const [postState, repostProfileState, reposts] = await Promise.all([
-      this.hydratePosts(postAndReplyRefs, ctx, {
-        posts: posts.merge(replies), // avoids refetches of posts
-      }),
-      this.hydrateProfiles(
-        [...repostUris.map(didFromUri), ...replyParentAuthors],
-        ctx,
-      ),
-      this.feed.getReposts(repostUris, ctx.includeTakedowns),
-    ])
+    const [postState, repostProfileState, reposts, rootThreadgates] =
+      await Promise.all([
+        this.hydratePosts(postAndReplyRefs, ctx, {
+          posts: posts.merge(replies), // avoids refetches of posts
+        }),
+        this.hydrateProfiles(
+          [...repostUris.map(didFromUri), ...replyParentAuthors],
+          ctx,
+        ),
+        this.feed.getReposts(repostUris, ctx.includeTakedowns),
+        this.feed.getThreadgatesForPosts(rootUris),
```

Comment:
Ah you know what, looking at this made me realize I was making redundant calls here. We already were hydrating all threadgates for root posts within `hydratePosts` since it's needed to power `viewer.replyDisabled`. So we can remove this line entirely.

Reply:
https://github.com/bluesky-social/atproto/pull/2675/commits/2447fa5ba9bd827c5097d7c73bdea94d7ba34a0e